### PR TITLE
xroar 1.5.5 (new formula)

### PR DESCRIPTION
**Pull request by felipebalbi**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ b] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
```
  clang  -I/opt/homebrew/Cellar/sdl2/2.30.6/include/SDL2 -D_THREAD_SAFE -g -O2   -o xroar xroar-ao.o xroar-auto_kbd.o xroar-becker.o xroar-blockdev.o xroar-breakpoint.o xroar-cart.o xroar-colourspace.o xroar-crc16.o xroar-crc32.o xroar-crclist.o xroar-dkbd.o xroar-events.o xroar-filter.o xroar-fs.o xroar-hexs19.o xroar-joystick.o xroar-keyboard.o xroar-logging.o xroar-machine.o xroar-module.o xroar-ntsc.o null/xroar-ui_null.o null/xroar-vo_null.o xroar-part.o xroar-path.o xroar-printer.o xroar-romlist.o xroar-screenshot.o xroar-serialise.o xroar-snapshot.o xroar-sound.o xroar-tape.o xroar-tape_cas.o xroar-ui.o xroar-vdg_palette.o xroar-vdisk.o xroar-vdrive.o xroar-vo.o xroar-vo_render.o xroar-xconfig.o xroar-xroar.o xroar-dragon.o xroar-coco3.o xroar-mc10.o mc6801/xroar-mc6801.o mc6809/xroar-hd6309.o mc6809/xroar-mc6809.o xroar-mc6821.o mc6847/xroar-font-6847.o mc6847/xroar-font-6847t1.o mc6847/xroar-mc6847.o xroar-mc6883.o tcc1014/xroar-font-gime.o tcc1014/xroar-tcc1014.o xroar-deltados.o xroar-dragondos.o xroar-gmc.o xroar-ide.o xroar-idecart.o xroar-mooh.o xroar-mpi.o xroar-nx32.o xroar-orch90.o xroar-rsdos.o xroar-sn76489.o xroar-spi65.o xroar-spi_sdcard.o xroar-wd279x.o xroar-main_unix.o  xroar-vo_opengl.o   sdl2/xroar-ao_sdl2.o sdl2/xroar-common.o sdl2/xroar-joystick_sdl2.o sdl2/xroar-keyboard_sdl2.o sdl2/xroar-ui_sdl2.o sdl2/xroar-vo_sdl2.o   macosx/xroar-filereq_cocoa.o macosx/xroar-ui_macosx.o sdl2/xroar-sdl_cocoa_keyboard.o    macosx/xroar-ao_macosx.o  null/xroar-ao_null.o    mc6801/xroar-mc6801_trace.o mc6809/xroar-mc6809_trace.o mc6809/xroar-hd6309_trace.o xroar-gdb.o  ../portalib/libporta.a -lm  -lGL    -L/opt/homebrew/Cellar/sdl2/2.30.6/lib -lSDL2  -framework Cocoa -framework Carbon    -framework CoreAudio    -lz  -pthread  
  Undefined symbols for architecture arm64:
    "_xroar_screenshot", referenced from:
        -[XRoarApplication do_set_state:] in xroar-ui_macosx.o
  ld: symbol(s) not found for architecture arm64
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  make[2]: *** [xroar] Error 1
  make[1]: *** [install] Error 2
  make: *** [install-recursive] Error 1
```

**Comment by felipebalbi** [comment]
> ```
>   clang  -I/opt/homebrew/Cellar/sdl2/2.30.6/include/SDL2 -D_THREAD_SAFE -g -O2   -o xroar xroar-ao.o xroar-auto_kbd.o xroar-becker.o xroar-blockdev.o xroar-breakpoint.o xroar-cart.o xroar-colourspace.o xroar-crc16.o xroar-crc32.o xroar-crclist.o xroar-dkbd.o xroar-events.o xroar-filter.o xroar-fs.o xroar-hexs19.o xroar-joystick.o xroar-keyboard.o xroar-logging.o xroar-machine.o xroar-module.o xroar-ntsc.o null/xroar-ui_null.o null/xroar-vo_null.o xroar-part.o xroar-path.o xroar-printer.o xroar-romlist.o xroar-screenshot.o xroar-serialise.o xroar-snapshot.o xroar-sound.o xroar-tape.o xroar-tape_cas.o xroar-ui.o xroar-vdg_palette.o xroar-vdisk.o xroar-vdrive.o xroar-vo.o xroar-vo_render.o xroar-xconfig.o xroar-xroar.o xroar-dragon.o xroar-coco3.o xroar-mc10.o mc6801/xroar-mc6801.o mc6809/xroar-hd6309.o mc6809/xroar-mc6809.o xroar-mc6821.o mc6847/xroar-font-6847.o mc6847/xroar-font-6847t1.o mc6847/xroar-mc6847.o xroar-mc6883.o tcc1014/xroar-font-gime.o tcc1014/xroar-tcc1014.o xroar-deltados.o xroar-dragondos.o xroar-gmc.o xroar-ide.o xroar-idecart.o xroar-mooh.o xroar-mpi.o xroar-nx32.o xroar-orch90.o xroar-rsdos.o xroar-sn76489.o xroar-spi65.o xroar-spi_sdcard.o xroar-wd279x.o xroar-main_unix.o  xroar-vo_opengl.o   sdl2/xroar-ao_sdl2.o sdl2/xroar-common.o sdl2/xroar-joystick_sdl2.o sdl2/xroar-keyboard_sdl2.o sdl2/xroar-ui_sdl2.o sdl2/xroar-vo_sdl2.o   macosx/xroar-filereq_cocoa.o macosx/xroar-ui_macosx.o sdl2/xroar-sdl_cocoa_keyboard.o    macosx/xroar-ao_macosx.o  null/xroar-ao_null.o    mc6801/xroar-mc6801_trace.o mc6809/xroar-mc6809_trace.o mc6809/xroar-hd6309_trace.o xroar-gdb.o  ../portalib/libporta.a -lm  -lGL    -L/opt/homebrew/Cellar/sdl2/2.30.6/lib -lSDL2  -framework Cocoa -framework Carbon    -framework CoreAudio    -lz  -pthread  
>   Undefined symbols for architecture arm64:
>     "_xroar_screenshot", referenced from:
>         -[XRoarApplication do_set_state:] in xroar-ui_macosx.o
>   ld: symbol(s) not found for architecture arm64
>   clang: error: linker command failed with exit code 1 (use -v to see invocation)
>   make[2]: *** [xroar] Error 1
>   make[1]: *** [install] Error 2
>   make: *** [install-recursive] Error 1
> ```

Odd, I built this on arm64 machine. I'll have a closer look later tonight.

**Comment by felipebalbi** [comment]
@SMillerDev , found a couple more dependencies that should have been listed. If these don't sort the problem out, I'm out of ideas. Builds fine on my M2 mac.

File: Formula/x/xroar.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+    depends_on "texinfo" => :build
+  end
+
+  depends_on "libpng"
+  depends_on "libsndfile"
+  depends_on "sdl2"
+
```
Comment (p-linnane) [inline Formula/x/xroar.rb:20]:
```suggestion

  uses_from_macos "zlib"

  on_linux do
    depends_on "alsa-lib"
  end

```

File: Formula/x/xroar.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  def install
+    system "./autogen.sh" if build.head?
+    system "./configure", "--disable-silent-rules", "--without-x", *std_configure_args
+    system "make", "install"
+  end
+
+  test do
+    assert_equal "XRoar #{version}",
+                 shell_output("#{bin}/xroar -V").split("\n").first
```
Comment (p-linnane) [inline Formula/x/xroar.rb:35]:
```suggestion
    assert_match version.to_s, shell_output(bin/"xroar -V")
```

Comment (p-linnane) [reply Formula/x/xroar.rb:35]: This is fine to include, but you'll need to add an additional test that actually verifies some functionality of the app.

Comment (felipebalbi) [reply Formula/x/xroar.rb:35]: That will be a bit hard as the app is an emulator of a retro computer. Testing actual functionality would involve downloading ROM images that might be copyrighted material in some countries. I just followed the example from atari800 formula.

Comment (p-linnane) [reply Formula/x/xroar.rb:35]: What about matching something from `xroar -config-print`? That won't involve ROM images and will ensure at least the binary is outputting expected information beyond version.

Comment (felipebalbi) [reply Formula/x/xroar.rb:35]: > What about matching something from `xroar -config-print`? That won't involve ROM images and will ensure at least the binary is outputting expected information beyond version.

I can certainly give it a go, but if e.g. new machines are added, that output might change in unexpected ways.

Comment (p-linnane) [reply Formula/x/xroar.rb:35]: No worries. If the output changes in a way that breaks the test, CI will catch it when we try to bump the version and we can fix it then.

Comment (felipebalbi) [reply Formula/x/xroar.rb]: @p-linnane, just pushed a new version matching the machines in the output of `-config-print`. Do you think it looks better?

**Review by p-linnane — APPROVED** [review]

**Pull request merged**